### ISSUE TITLE
[BUG] MlflowClient use a relative path to download_artifacts without using the absolute path in tracking_uri

### ISSUE THREAD

**Issue by you-n-g** [mlflow/mlflow#2816]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [X] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: 
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04.6 LTS
- **MLflow installed from (source or binary)**:
source
- **MLflow version (run ``mlflow --version``)**:
mlflow, version 1.8.0
- **Python version**:
Python 3.7.5
- **npm version, if running the dev UI**:
6.9.0
- **Exact command to reproduce**:
![image](https://user-images.githubusercontent.com/465606/81689736-2200d180-948d-11ea-8d7b-220f4bce2990.png)

### Describe the problem
Describe the problem clearly here. Include descriptions of the expected behavior and the actual behavior.
Run MlflowClient.download_artifacts in a different directory with the rightly setting the absolute tracking_uri with `mlflow.set_tracking_uri`

The MlflowClient could find the right run_info, but it use a relative path to load the the artifacts.

### Code to reproduce issue
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
![image](https://user-images.githubusercontent.com/465606/81689717-1e6d4a80-948d-11ea-9242-02fb4ba1b50e.png)

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
![image](https://user-images.githubusercontent.com/465606/81689351-c8000c00-948c-11ea-81aa-a10e5572cd39.png)

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

**Comment by smurching** [comment]
Hi @you-n-g, thanks for reporting this issue. The problem here seems to be that the run was created under an experiment whose artifact root is a relative path on your local filesystem, rather than an absolute path. Do you know how the experiment in this case (with ID '20') was created?

I tried to create an experiment against my local `mlruns` directory using `mlflow experiments create -n <name>` (using MLflow 1.8.0) and found that the artifact path was indeed an absolute path, so further instructions or ideas on how your experiment was created to use a relative artifact path would be appreciated - thanks!

**Comment by you-n-g** [comment]
@smurching  
Do you mean that the `artifact_location` is a relative path when you say it is a relative path?
![image](https://user-images.githubusercontent.com/465606/82279465-37ab5500-99bf-11ea-8efe-da56bcdef71e.png)

I created the folder by clicking on the web interface, which is run by  `mlflow ui -h <hostname>`

![image](https://user-images.githubusercontent.com/465606/82279644-9ec90980-99bf-11ea-8ada-0170f32fb738.png)

**Comment by smurching** [comment]
@you-n-g that's right - this actually seems like a bug in MLflow. Namely, we should update [this line](https://github.com/mlflow/mlflow/blob/939b494d9cccbb32999439fa744076f5fad28749/mlflow/store/tracking/__init__.py#L12) to ensure that we compute an absolute path for the default artifact URI (i.e. replace it with `os.path.abspath(DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH)`). Would you be interested in opening a PR? Happy to review a fix.

**Comment by you-n-g** [comment]
I've encoutered some problems when deploying mlflow develop environment  

1. I install mlflow by `python setup.py develop`
2. I follow the [instructures](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#running-the-javascript-dev-server)
![image](https://user-images.githubusercontent.com/465606/87365609-41110200-c5a9-11ea-81fe-0e573da03ec8.png)
![image](https://user-images.githubusercontent.com/465606/87365615-4706e300-c5a9-11ea-88d1-8d102cf291ca.png)

3. But I still get the error info page
![image](https://user-images.githubusercontent.com/465606/87365649-5a19b300-c5a9-11ea-9959-3e2447a38633.png)
